### PR TITLE
DOC: Add a coding style guideline.

### PR DESCRIPTION
**Pull request by jhlegarreta**
Add a coding style guideline mainly aimed at establishing some criteria
for the example and rst documentation files.

### CODE REVIEW

**Comment by jhlegarreta** [comment]
@Garyfallidis  @arokem @skoudoro @dmreagan Please review the PR. It is still a reduced style guide, to be completed as I learn DIPY or revise more code, we receive suggestions, etc.

File: doc/devel/coding_style_guideline.rst [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+
+-------------
+Documentation
+-------------
+
+DIPY relies on a set of files that serve as an additional pool of information
+for users and contributors:
+- reStructuredText files (``*.rst``): the files located in the ``<./doc>``
+folder structure contain informaton about theoretical backgrounds of DIPY,
```
Comment (skoudoro) [inline doc/devel/coding_style_guideline.rst]: informaton -> information

File: doc/devel/coding_style_guideline.rst [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+DIPY core
+-----------
+
+DIPY has adopted the standard Python `PEP8
+<https://www.python.org/dev/peps/pep-0008/>_` style to ensure the
+readability and consistency across the toolkit. The PEP8 style is enforced
+automatically when requesting to push to DIPY. However, as a contributor to
+DIPY, you should try to ensure that your code, including its comments
+
+
```
Comment (skoudoro) [inline doc/devel/coding_style_guideline.rst]: I think you can add that we follow [the numpy coding style](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt)

Comment (skoudoro) [inline doc/devel/coding_style_guideline.rst]: and you can cite the Numpy detailed example [here](https://github.com/numpy/datetime/blob/master/doc/EXAMPLE_DOCSTRING.txt)

Comment (jhlegarreta) [inline doc/devel/coding_style_guideline.rst]: The example refers to a C-function; I guess you meant [example.py](https://github.com/numpy/numpy/blob/master/doc/example.py), right?

Comment (jhlegarreta) [inline doc/devel/coding_style_guideline.rst]: @skoudoro thanks for the review and for having spot my slip-ups! See whether commit ca65ba8 addresses the comments.

**Comment by jhlegarreta** [comment]
BTW, should this be linked from or to [CONTRIBUTING.md]( https://github.com/nipy/dipy/blob/master/CONTRIBUTING.md)? Or else, should we integrate it into there? The numpy and PEP8 paragraphs overlap.

**Comment by arokem** [comment]
I think best to move duplication from CONTRIBUTING into this guide, and then link from the guide to the documentation page.

**Comment by jhlegarreta** [comment]
@arokem Done. Please review the new patch set.

I've included a line width limit criterion for examples and `.rst` files; I don't know whether that's written in stone, but it's something I've observed across the files. Please, comment appropriately, as I'd like to know whether this has a particular (historic/Unix-like editor?) reason.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -51,39 +51,18 @@ fully covered are very welcome!
...
 write code that works on both versions.
 
 ### Code style
 
-Code contributions should be formatted according to the
-[PEP8](https://www.python.org/dev/peps/pep-0008/) style guide. There are
-[software systems](https://pypi.python.org/pypi/pep8) that will check your code
-for PEP8 compliance, and most text editors can be configured to check the
-compliance of your code with PEP8.
+Code contributions should be formatted according to the [DIPY Coding Style Guideline](./doc/devel/coding_style_guideline.rst).
```
Comment (arokem) [inline CONTRIBUTING.md]: Eventually, once the style guide gets to the website, we should link to the website docs, instead?

File: doc/devel/coding_style_guideline.rst [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+choices, helps avoiding obscure places, and is a way to allow other members
+*decode* it with less effort.
+* **Language**: the code must be written in English. Norms and spelling should
+be abided by.
+
+------------
+Coding style
+------------
+
+DIPY follows the numpy_ coding style, described in their
```
Comment (arokem) [inline doc/devel/coding_style_guideline.rst]: This is more about docstrings than about "coding"

Comment (arokem) [inline doc/devel/coding_style_guideline.rst]: In particular, I would move this to the "documentation" section below

File: doc/devel/coding_style_guideline.rst [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+-------------
+Documentation
+-------------
+DIPY uses `Sphinx <http://www.sphinx-doc.org/en/stable/index.html>`_ to generate
+documentation. We welcome contributions of examples, and suggestions for changes
+in the documentation, but please make sure that changes that are introduced
+render properly into the HTML format that is used for the DIPY website.
+
+When documenting functions, modules, class or methods we follow the
```
Comment (arokem) [inline doc/devel/coding_style_guideline.rst]: I see -- it's already here! No need for that sentence in the previous section, IMO.

**Comment by arokem** [comment]
Great work here, @jhlegarreta! I had just a couple of small comments.

File: CONTRIBUTING.md (hunk 3 above)
Comment (jhlegarreta) [inline CONTRIBUTING.md]: True. I guess we'll need to wait until it gets to the website to get the URL. I set a reminder for this.

File: doc/devel/coding_style_guideline.rst (hunk 4 above)
Comment (jhlegarreta) [inline doc/devel/coding_style_guideline.rst]: Makes sense. Removed the sentence from the Coding style section, and reformatted the one in the Documentation section in c51e652.

File: doc/devel/coding_style_guideline.rst (hunk 5 above)
Comment (jhlegarreta) [inline doc/devel/coding_style_guideline.rst]: Addressed in c51e562,

**Comment by jhlegarreta** [comment]
@arokem Thanks for the review.

OK, I identified the source for the maximum line width limitation: PEP8 and numpy docstring both establish a limitation. So removed the sentence I introduced in the last commit. Although they're different (72 vs. 75), I guess we don't have strong preferences.

Should we establish a hard criterion, please let me know (since the current files have generally been limited to 80...).

**Comment by jhlegarreta** [comment]
The failing build in Travis seems to be unrelated to the patch set.  I ignore the meaning of the `USE_PRE` flag, but I've seen that other builds have the build marked as `Allowed Failures` (e.g. https://travis-ci.org/nipy/dipy/builds/214169250).

**Comment by jhlegarreta** [comment]
Rebased 7c809e6 onto master to see if the errors go away (although the branch simply modifies a Markdown file, and AFAIK there is not hook/check on .md files).

**Comment by skoudoro** [comment]
Thanks for that @jhlegarreta. I think it is ready to go. @arokem is it ok for you?

I do not think that we need a hard criterion for the maximum line width limitation.

Concerning the paths, written as <./doc/devel>, its deserve its own new PR :-)

**Comment by arokem** [comment]
Yep. Looks good. Merging!

**Pull request merged**